internal/settings/settings.go

```diff
@@ -60,77 +57,58 @@ const (
 	ConfigDScheme       = "splunk.configd"
 )
 
-type Settings interface {
```

Comment:
> And, as an opportunistic simplification, this change replaces an unnecessary Settings interface with a struct.

The aim of this interface was to help clarify the intended consumer usage or available settings while hiding all the underlying flag/state management that was scattered throughout `main.go`. What are we gaining by using the exported internal struct directly?

Reply:
Any IDE or editor can show all available methods and point you right to the implementation, which is much better than pointing to an interface declaration that force you to look around for the actual implementation after that. Also, all the methods will be listed the same way as they are in the go doc https://pkg.go.dev/github.com/signalfx/splunk-otel-collector@v0.66.0/internal/settings#Settings. So would disagree that using interfaces for one struct makes developer live any better.

Reply:
> It's more for maintainability 

IMHO adding unnecessary interfaces reduces maintainability

> establish a clear convention for how future settings should be added

I think it's rather an unnecessarily complicated convention for such a simple interface. Not sure if there is even a reason to expose them as methods... If they are all just getters using data from the struct that was set during initialization, we can even expose them as precalculated struct fields. That will put them all together in one place as it was with the interface. And it'll help to avoid extra work if they would be used more than once in the future. What do you think?

Reply:
Not to get too caught up in a dispute of personal preference, but they were interfaces because interfaces are collections of methods and not fields. The accessors do the computing and return the setting value, but could definitely be improved.  They were exposed via the settings interface to encapsulate the ~legacy flag helpers while migrating to a central setting store to prevent breaking changes.

Processing them w/ initial parsing stages in `New()` and using exported fields is acceptable to me and would address their idempotence issues.

Reply:
> Processing them w/ initial parsing stages in New() and using exported fields is acceptable to me and would address their idempotence issues.

Sounds good, I was thinking about doing that but didn't want to go too far. I can do it in this or separate PR